src/axom/multimat/docs/sphinx/multimat_field_concepts.rst

```diff
@@ -0,0 +1,118 @@
+|                    | For scalars *ncomponents* is 1 so the field length is    |
+|                    | ncells.                                                  |
++--------------------+----------------------------------------------------------+
+| PER_MAT            | The field contains nmats*ncomponents values and there    |
+|                    | are ncomponents values per material. This mapping allows |
+|                    | fields to be defined over the entire material region and |
+|                    | any cell that uses the material inherits the per-material|
+|                    | field value, allowing for very compact storage of        |
+|                    | material-level properties.                               |
++--------------------+----------------------------------------------------------+
+| PER_CELL_MAT       | The field contains up to ncells*nmats*ncomponents (for   |
+|                    | dense storage). This mapping allows materials within a   |
+|                    | cell to have their own unique values, which makes them   |
+|                    | useful for tracking data at a sub-cell level.            |
++--------------------+----------------------------------------------------------+
+
+#######################
+Data Layout
+#######################
+
+Simulation codes contain a variety of algorithms that may have a preference for how
+data are arranged in memory to ensure good performance. MultiMat supports
+fields with a ``PER_CELL_MAT`` mapping and there are two ways to organize such data.
+Fields are said to be **Cell-Dominant** (``CELL_DOM``) if they are stored such that
+each cell stores all of its material data to memory before proceeding to data for
+the next cell. Fields are **Material-Dominant** (``MAT_DOM``) if the data for all
+cells that use the material is stored before proceeding to the next material.
+The data layout for multi-material data can be thought of as 2 nested for-loops where
+the outer loop is the dominant loop. For example, if iterating over materials and
+then cells, the data are stored using ``MAT_DOM`` layout.
+
++--------------------+----------------------------------------------------------+
+| DataLayout         | Meaning                                                  |
++====================+==========================================================+
+| CELL_DOM           | Data are stored for each cell and then for each material |
+|                    | like this *(c=cell, m=material)*:                        |
+|                    |                                                          |
+|                    | ``{c0m0, c0m1, c0m2, ..., c1m0, c1m1, c1m2, ...}``       |
++--------------------+----------------------------------------------------------+
+| MAT_DOM            | Data are stored for each material and then for each cell |
+|                    | like this *(c=cell, m=material)*:                        |
```

Comment:
I'd prefer it if you wrote "*(m=material, c=cell)*".  I think it's clearer because you write below "m0c0, m0c1, m0c2, ..."

Reply:
Changed.